test/ValidatorSet/ValidatorSet.test.ts

```diff
@@ -356,18 +354,110 @@ describe("ValidatorSet", function () {
             .addToWhitelist([this.signers.validators[0].address, this.signers.validators[1].address])
         ).to.not.be.reverted;
 
-        expect((await validatorSet.validators(this.signers.validators[0].address)).whitelisted).to.be.true;
-        expect((await validatorSet.validators(this.signers.validators[1].address)).whitelisted).to.be.true;
+        expect((await validatorSet.validators(this.signers.validators[0].address)).status).to.be.equal(
+          VALIDATOR_STATUS.Whitelisted
+        );
+        expect((await validatorSet.validators(this.signers.validators[1].address)).status).to.be.equal(
+          VALIDATOR_STATUS.Whitelisted
+        );
+      });
+
+      it("owner should not whitelist user that is already whitelisted", async function () {
+        const { validatorSet } = await loadFixture(this.fixtures.initializedValidatorSetStateFixture);
+
+        await expect(
+          validatorSet
+            .connect(this.signers.governance)
+            .addToWhitelist([this.signers.validators[0].address, this.signers.validators[1].address])
+        ).to.not.be.reverted;
+
+        await expect(
+          validatorSet
+            .connect(this.signers.governance)
+            .addToWhitelist([this.signers.validators[0].address, this.signers.validators[1].address])
+        ).to.be.revertedWith("Previously whitelisted.");
+
+        expect((await validatorSet.validators(this.signers.validators[0].address)).status).to.be.equal(
```

Comment:
Since we have verified that the whitelist works fine in the previous test, we don't have to do it again here. So, we can leave the function until the expected revert.